third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -2480,18 +2480,6 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
         ) = &args[1].value
         {
             let field_name = self.symbol_pool().make(name.value.as_str());
-            let constraint =
-                Constraint::SomeStruct([(field_name, expected_type.clone())].into_iter().collect());
-            if let Err(e) = self.subs.eval_constraint(
-                &self.unification_context,
-                loc,
-                expected_type,
```

Comment:
The problem is that this expected type here is wrong. Instead of the expected type of the expression (which is the struct we are updating) we should put in here the expected type of the value of the field, which is the type of the last value parameter of update_field.

Reply:
PS. Adding a few test cases to tests/checking/specs might also be helpful. We should test field wrongly named, field wrong type, etc.

Reply:
done